frontend/pkg/frontend/middleware_logging.go

```diff
@@ -77,39 +81,77 @@ func MiddlewareLoggingPostMux(w http.ResponseWriter, r *http.Request, next http.
 	ctx := r.Context()
 	logger := LoggerFromContext(ctx)
 
-	attrs := getLogAttrs(r)
-	logger = slog.New(logger.Handler().WithAttrs(attrs))
-	r = r.WithContext(ContextWithLogger(ctx, logger))
+	attrs := &attributes{
+		subscriptionID: r.PathValue(PathSegmentSubscriptionID),
+		resourceGroup:  r.PathValue(PathSegmentResourceGroupName),
+		resourceName:   r.PathValue(PathSegmentResourceName),
+	}
+	attrs.addToCurrentSpan(ctx)
+	ctx = ContextWithLogger(ctx, attrs.extendLogger(logger))
+	r = r.WithContext(ctx)
 
 	next(w, r)
 }
 
-// getLogAttrs returns the additional Logging attributes based on the wildcards
-// from the matched pattern.
-func getLogAttrs(r *http.Request) []slog.Attr {
+type attributes struct {
+	subscriptionID string
+	resourceGroup  string
+	resourceName   string
+}
+
+func (a *attributes) resourceID() string {
+	if a.subscriptionID == "" || a.resourceGroup == "" || a.resourceName == "" {
+		return ""
+	}
+
+	return fmt.Sprintf(
+		"/subscriptions/%s/resourcegroups/%s/providers/%s/%s",
+		a.subscriptionID,
+		a.resourceGroup,
+		api.ClusterResourceType,
+		a.resourceName,
+	)
+}
+
+// extendLogger returns a new logger with additional Logging attributes based
+// on the wildcards from the matched pattern.
+func (a *attributes) extendLogger(logger *slog.Logger) *slog.Logger {
 	var attrs []slog.Attr
 
-	subscriptionID := r.PathValue(PathSegmentSubscriptionID)
-	if subscriptionID != "" {
-		attrs = append(attrs, slog.String("subscription_id", subscriptionID))
+	if a.subscriptionID != "" {
+		attrs = append(attrs, slog.String("subscription_id", a.subscriptionID))
 	}
 
-	resourceGroup := r.PathValue(PathSegmentResourceGroupName)
-	if resourceGroup != "" {
-		attrs = append(attrs, slog.String("resource_group", resourceGroup))
+	if a.resourceGroup != "" {
+		attrs = append(attrs, slog.String("resource_group", a.resourceGroup))
+	}
+
+	if a.resourceName != "" {
+		attrs = append(attrs, slog.String("resource_name", a.resourceName))
+	}
+
+	if resourceID := a.resourceID(); resourceID != "" {
+		attrs = append(attrs, slog.String("resource_id", resourceID))
+	}
+
+	return slog.New(logger.Handler().WithAttrs(attrs))
+}
+
+func (a *attributes) addToCurrentSpan(ctx context.Context) {
+	span := trace.SpanFromContext(ctx)
+
+	var attrs []attribute.KeyValue
+	if a.subscriptionID != "" {
+		attrs = append(attrs, attribute.String("aro.subscription_id", a.subscriptionID))
```

Comment:
Are we sticking to `subscription_id` or `subcription.id`? I can't remember if we had a conversion on the ID convention or not. Remind me if we did.

Reply:
good point and I don't remember that we decided explicitly for one or the other.

Reply:
but since we already have `aro.correlation.id`, it's probably best to stick with the '.' approach. I'll update.

Reply:
ok after thinking a bit more, I have taken a mixed approach. We can discuss more next week (nothing urgent here).